src/components/utilityFunctions.ts

```diff
@@ -85,4 +85,13 @@ export const filterObject = (
   return Object.fromEntries(filtered)
 }
 
+export const crawlFileSystem = async (newPath: string, checkFile: (newPath: string) => void) => {
+  const files = fs.readdirSync(newPath)
+  for (const file of files) {
+    const subPath = path.join(newPath, file)
+    if (fs.statSync(subPath).isDirectory()) crawlFileSystem(subPath, checkFile)
+    else checkFile(subPath)
+  }
+}
+
```

Comment:
Cool, looks good. Can you just give the input parameters slightly more accurate names, e.g.
- `newPath` -- why not just `path` (there's nothing particularly "new" about it)
- `checkFile` -- we don't know what this function could be -- any operation on a file path, so shouldn't have such a specific name. Maybe just `fileOperation`, or `handleFile`? (I know you called it `checkFile` for the actual function we're using in this case -- that's fine, but no need for the general parameter to have the same specific name)